src/services/AccommodationService.js

```diff
@@ -0,0 +1,86 @@
+import { likes, accommodations } from '../database/models';
+import CommonQueries from './CommonQueries';
+
+/**
+ * @export
+ * @class AccommodationService
+ */
+class AccommodationService {
+  /**
+ * user POST likes/dislikes accommodation facility
+ * @static
+ * @param {object} req pass request body
+ * @memberof AccommodationService
+ * @returns {object} either an error or data
+ */
+  static async addAccommodationLike(req) {
+    let setObject, displayMessage;
+    const { like } = req.query;
+    const { accommodationId } = req.params;
+    const { id } = req.user;
+
+    const accommodation = await CommonQueries.findOne(accommodations, { where: { id: accommodationId } });
+    const isLikedOrDisliked = await CommonQueries.findOne(likes, { where: { userId: id } });
+
+    if (isLikedOrDisliked) {
+      const { liked, disliked } = isLikedOrDisliked.dataValues;
+
+      if (like && liked && !disliked) {
+        setObject = { liked: false };
+        displayMessage = 'You unliked this accommodation!';
+      }
+      if (!like && !liked && disliked) {
+        setObject = { disliked: false };
+        displayMessage = 'You un-disliked this accommodation!';
+      }
+      if (!like && liked && !disliked) {
+        setObject = { disliked: true, liked: false };
+        displayMessage = 'You disliked this accommodation!';
+      }
+      if (!liked && !disliked) {
+        setObject = { liked: like, disliked: !like };
+        displayMessage = like ? 'You liked this accommodation!' : 'You disliked this accommodation!';
+      }
```

Comment:
@victorkarangwa4 I think we can find a way to probably improve this.
I'll try to find one and share with you.